### PR TITLE
Variable selection list enhancement

### PR DESCRIPTION
**Pull request by TomSendrovich**
Thank you for your contribution.
Before submitting this PR, please make sure:
- [ ] PR description and commit message should describe the changes done in this PR
- [ ] Verify the PR is point to correct branch i.e. Release or Beta branch if the code fix is for specific release , else point it to master
- [ ] Latest Code from master or specific release branch is merged to your branch
- [ ] No unwanted\commented\junk code is included
- [ ] No new warning upon build solution
- [ ] Existing unit test cases are passed
- [ ] New Unit tests are added for your development
- [ ] Sanity Tests are successfully executed for Existing Functionality
- [ ] After creating pull request there should not be any conflicts
- [ ] Resolve codacy comments
- [ ] Builds and checks are passed before PR is sent for review
- [ ] Resolve code review comments

### CODE REVIEW

File: Ginger/Ginger/Reports/ValidationRules/ValidateNotContainSpecificChar.cs [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+using System.Threading.Tasks;
+using System.Windows.Controls;
+
+namespace Ginger.Reports.ValidationRules
+{
+    public class ValidateNotContainSpecificChar : ValidationRule
+    {
+        private string _Message = string.Empty;
+
+        public ValidateNotContainSpecificChar(string message = "Variable name can't contain a comma (',')")
```
Comment (MeniKadosh1) [inline Ginger/Ginger/Reports/ValidationRules/ValidateNotContainSpecificChar.cs:15]: i didn't got why to set the error message in the constructor and not to get in it the char to validate?
Validation rule is called 'ValidateNotContainSpecificChar' so it needs to be generic and validate any char and not only comma

File: Ginger/Ginger/Variables/VariableEditPage.xaml.cs [hunk 2]
```diff
@@ -74,6 +75,8 @@ public VariableEditPage(VariableBase v, Context context, bool setGeneralConfigsA
             mContext = context;
             GingerCore.GeneralLib.BindingHandler.ObjFieldBinding(xTypeLbl, Label.ContentProperty, mVariable, nameof(VariableBase.VariableType), BindingMode: BindingMode.OneWay);
             GingerCore.GeneralLib.BindingHandler.ObjFieldBinding(xVarNameTxtBox, TextBox.TextProperty, mVariable, nameof(VariableBase.Name));
+            xVarNameTxtBox.AddValidationRule(new ValidateNotContainSpecificChar());
```
Comment (MeniKadosh1) [inline Ginger/Ginger/Variables/VariableEditPage.xaml.cs:78]: here need to provide the comma as a parameter

File: Ginger/GingerCoreCommon/External/Configurations/VariableConfiguration.cs [hunk 3]
```diff
@@ -0,0 +1,29 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Amdocs.Ginger.Repository;
+
```
Comment (MeniKadosh1) [inline Ginger/GingerCoreCommon/External/Configurations/VariableConfiguration.cs:7]: what this class is for?

File: Ginger/GingerCoreCommon/WorkSpaceLib/Solution.cs [hunk 4]
```diff
@@ -298,6 +299,9 @@ public ExecutionLoggerConfiguration LoggerConfigurations
         [IsSerializedForLocalRepository]
         public VRTConfiguration VRTConfiguration { get; set; } = new VRTConfiguration();
 
+        [IsSerializedForLocalRepository]
+        public VariableConfiguration VariableConfiguration { get; set; } = new VariableConfiguration();
```
Comment (MeniKadosh1) [inline Ginger/GingerCoreCommon/WorkSpaceLib/Solution.cs:303]: for what is this?

File: Ginger/GingerCoreNET/ValueExpression/ValueExpessionGeneralFunctions.cs [hunk 5]
```diff
@@ -50,6 +51,17 @@ public string GetGUID()
...
         }
 
+        [ValueExpressionFunctionAttribute]
+        [ValueExpressionFunctionDescription("Get Clipboard Text")]
+        [ValueExpressionFunctionExpression("{Function Fun=GetClipboardText()}")]
+        [ValueExpressionFunctionCategory("Data")]
+        [ValueExpressionFunctionSubCategory("Functions")]
+        public string GetClipboardText()
+        {
+            return WorkSpace.Instance.OSHelper.GetClipboardText();
```
Comment (MeniKadosh1) [inline Ginger/GingerCoreNET/ValueExpression/ValueExpessionGeneralFunctions.cs:61]: we said to push it later once we'll find solution for it, no?

File: Ginger/Ginger/Reports/ValidationRules/ValidateNotContainSpecificChar.cs [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+        {
+            _Message = message;
+
```
Comment (MeniKadosh1) [inline Ginger/Ginger/Reports/ValidationRules/ValidateNotContainSpecificChar.cs:18]: let's also add a validation rule that name do not start or end with space

File: Ginger/GingerUtils/Properties/Resources.resx [hunk 7]
```diff
@@ -0,0 +1,101 @@
...
+		that is mostly human readable. The generation and parsing of the 
+		various data types are done through the TypeConverter classes 
+		associated with the data types.
+
+		Example:
+
+		... ado.net/XML headers & schema ...
+		<resheader name="resmimetype">text/microsoft-resx</resheader>
+		<resheader name="version">1.3</resheader>
+		<resheader name="reader">System.Resources.ResXResourceReader, System.Windows.Forms, ...</resheader>
```
Comment (MeniKadosh1) [inline Ginger/GingerUtils/Properties/Resources.resx:18]: why this been pushed?

File: Ginger/GingerUtils/Properties/Resources.Designer.cs [hunk 8]
```diff
@@ -0,0 +1,63 @@
...
+
+namespace GingerUtils.Properties {
+    using System;
+    
+    
+    /// <summary>
+    ///   A strongly-typed resource class, for looking up localized strings, etc.
+    /// </summary>
+    // This class was auto-generated by the StronglyTypedResourceBuilder
+    // class via a tool like ResGen or Visual Studio.
```
Comment (MeniKadosh1) [inline Ginger/GingerUtils/Properties/Resources.Designer.cs:19]: why this been pushed?

**Review by MeniKadosh1 — CHANGES_REQUESTED** [review]

File: Ginger/GingerCoreNET/ValueExpression/ValueExpessionGeneralFunctions.cs (hunk 5 above)
Comment (TomSendrovich) [reply Ginger/GingerCoreNET/ValueExpression/ValueExpessionGeneralFunctions.cs:61]: I only added the basic implementation, So I will comment this code so I can easily come back to implement it later

File: Ginger/GingerUtils/Properties/Resources.resx (hunk 7 above)
Comment (TomSendrovich) [reply Ginger/GingerUtils/Properties/Resources.resx:18]: I don't know, I haven't changed this file at all

File: Ginger/GingerCoreCommon/External/Configurations/VariableConfiguration.cs (hunk 3 above)
Comment (TomSendrovich) [reply Ginger/GingerCoreCommon/External/Configurations/VariableConfiguration.cs]: I followed VRTCOnfigurations and this is what they have done there. I deleted this class and it works perfectly fine without it.

File: Ginger/GingerCoreCommon/WorkSpaceLib/Solution.cs (hunk 4 above)
Comment (TomSendrovich) [reply Ginger/GingerCoreCommon/WorkSpaceLib/Solution.cs]: I deleted this

File: Ginger/GingerUtils/Properties/Resources.Designer.cs (hunk 8 above)
Comment (TomSendrovich) [reply Ginger/GingerUtils/Properties/Resources.Designer.cs:19]: I deleted this

**Comment by MeniKadosh1** [comment]
no need to merge this.

**Pull request closed without merging**